src/utils.cpp

```diff
@@ -944,44 +943,7 @@ unsigned int utils::gentabs(const std::string& str)
  * exist. */
 int utils::mkdir_parents(const std::string& p, mode_t mode)
 {
-	int result = -1;
-
-	/* Have to copy the path because we're going to modify it */
-	std::vector<char> pathname(p.cbegin(), p.cend());
-	pathname.push_back('\0');
-	/* This pointer will run through the whole string looking for '/'.
-	 * We move it by one if path starts with slash because if we don't, the
-	 * first call to access() will fail (because of empty path) */
-	char* curr = pathname.data() + (pathname[0] == '/' ? 1 : 0);
-
-	while (*curr) {
-		if (*curr == '/') {
-			*curr = '\0';
-			result = mkdir(pathname.data(), mode);
-			if (result != 0) {
-				if (errno == EEXIST) {
-					result = 0;
-				} else {
-					break;
-				}
-			}
-			*curr = '/';
-		}
-		curr++;
-	}
-
-	if (result == 0) {
-		result = mkdir(p.c_str(), mode);
-
-		// It's not an error if the directory already exists. This happens when
-		// `p` ended with a slash, in which case the loop above creates all the
-		// path components.
-		if (result == -1 && errno == EEXIST) {
-			result = 0;
-		}
-	}
-
-	return result;
+	return rs_mkdir_parents(p.c_str(), (int) mode);
```

Comment:
`(int)mode` is a C-style cast, and it's very powerful (something like Rust's `mem::transmute`). It's better to use one of C++'s more restricted casts, in this case `static_cast<std::uint32_t>(mode)`.

I'm actually not sure if `mode_t` is always 32 bits, but upcasting doesn't hurt, and an attempt in downcasting from 64 bits or whatever will give us a compilation error—all acceptable outcomes.

Reply:
Maybe I should leave a comment for future reference?

Reply:
What would the comment say? If this code fails to compile, I think the error will be enugh, saying something like `no matching conversion for static_cast from 'mode_t' (aka 'foo bar') to 'std::uint32_t' (aka 'baz quux')`.